### ISSUE TITLE
nonzero layer test  

### ISSUE THREAD

**Issue by lix19937** [lix19937/tensorrt-insight#8]
tensor mask op，as example：x = x[x.sum(dim=-1) > 0]    
torch.where(condition), as example: torch.where(x>0)    
torch.nonzero(x)

**Comment by lix19937** [comment]
对于 输入为fixed shape，而内部出现 nonzero 操作的一般认为是`局部dynamic`， currently skipped for dynamic shapes

**Comment by lix19937** [comment]
look into the model to check if the NonZero can be replaced, sometimes onnx exporter just try to generate some indices to embedding layer, it will also use NoneZero, in this case it can be replaced with Range

**Comment by lix19937** [comment]
`data dependent shape` be not supported by 8.5.x

**Comment by lix19937** [comment]
https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-853/api/c_api/classnvinfer1_1_1_i_network_definition.html#ae4a4d21b214bb82d5d441ba03590f726       https://github.com/NVIDIA/TensorRT/issues/2285
